pkg/controllers/execution/execution_controller.go

```diff
@@ -185,6 +185,7 @@ func (c *Controller) syncToClusters(clusterName string, work *workv1alpha1.Work)
 	for _, manifest := range work.Spec.Workload.Manifests {
 		workload := &unstructured.Unstructured{}
 		err := workload.UnmarshalJSON(manifest.Raw)
+		reference, _ := helper.GenEventRef(c.Client.Scheme(), workload)
```

Comment:
Probably we shouldn't ignore the error here. 

Maybe we can introduce a method to the `execution controller`, like:
```golang
func (c *Controller) Eventf(object *unstructured.Unstructured, eventtype, reason, messageFmt string, args ...interface{}) {
	ref, err := helper.GenEventRef(...)
	if err != nil {
		fmt.Errorf("ignored event(%s) as failed to build event reference: %s", reason, err)
	}
	
	c.EventRecorder.Eventf(ref, eventtype, reason, messageFmt, args...)
}
```

Reply:
Okay